kubernetes/resource_kubernetes_job_v1.go

```diff
@@ -48,6 +51,78 @@ func resourceKubernetesJobV1() *schema.Resource {
 	}
 }
 
+func resourceKubernetesJobV1CustomizeDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{}) error {
+	if d.Id() == "" {
+		log.Printf("[DEBUG] Resource ID is empty, resource not created yet.")
+		return nil
+	}
+
+	// Retrieve old and new TTL values as strings
+	oldTTLRaw, newTTLRaw := d.GetChange("spec.0.ttl_seconds_after_finished")
+
+	var oldTTLStr, newTTLStr string
+
+	if oldTTLRaw != nil {
+		oldTTLStr, _ = oldTTLRaw.(string)
+	}
+	if newTTLRaw != nil {
+		newTTLStr, _ = newTTLRaw.(string)
+	}
+
+	oldTTLInt, err := strconv.Atoi(oldTTLStr)
+	if err != nil {
+		oldTTLInt = 0
+	}
+	newTTLInt, err := strconv.Atoi(newTTLStr)
+	if err != nil {
+		newTTLInt = 0
+	}
+
+	conn, err := meta.(KubeClientsets).MainClientset()
+	if err != nil {
+		return err
+	}
+
+	namespace, name, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Check if the Job exists
+	_, err = conn.BatchV1().Jobs(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Job is missing
+			if oldTTLInt >= 0 {
+				if oldTTLInt != newTTLInt {
+					// TTL value changed; force recreation
```

Comment:
I actually don't think you need to put the logic for forcing recreation when the ttl changes here. You just need to set `ForceNew: true` here: 

https://github.com/hashicorp/terraform-provider-kubernetes/blob/69263f8d4b7eb1b1227dbd7ee985fd107d195a54/kubernetes/schema_job_spec.go#L235-L247

Reply:
In ttl_seconds_after_finished attr, the field ForceNew is now set to true!